### PR TITLE
docs revamp: Add new documentation to sidebar index

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

This includes the new documentation to the sidebar index.

**Special notes for your reviewer:**

In order to run the documentation locally using
```
$ cd docs
$ make docs
```
After the docker image is running, the service should be available at http://localhost:3002

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/auth/_index.md [hunk 1]
```diff
@@ -1 +1,14 @@
+---
+title: Authentication setup
+menuTitle: Authentication setup
+description: Documentation reference for identity providers and authentication services.
+aliases:
```
Comment (chri2547) [inline docs/sources/auth/_index.md:5]: Has this file moved? If not, then we don't need an alias at the moment. Said another way, aliases should only document where the file used to exist and do not need to document its current location.

**Review by chri2547 — COMMENTED** [review]
A couple of comments. Thank you!

File: docs/sources/auth/_index.md [hunk 2]
```diff
@@ -1 +1,14 @@
+---
+title: Authentication setup
```
[the remark this answers is not in the document]
Comment (linoman) [reply docs/sources/auth/_index.md:2]: Changed.

File: docs/sources/auth/_index.md (hunk 1 above)
Comment (linoman) [reply docs/sources/auth/_index.md:5]: Changed.

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**